controllers/package_controller.go

```diff
@@ -114,9 +115,11 @@ func RegisterPackageReconciler(mgr ctrl.Manager) (err error) {
 }
 
 func createSecretAuth(cfg *rest.Config) (auth.Authenticator, error) {
-	scheme := runtime.NewScheme()
-	cfg.GroupVersion = &api.GroupVersion
-	cfg.NegotiatedSerializer = serializer.NewCodecFactory(scheme)
```

Comment:
This new approach has all sorts of hard coded strings. No better way?

Reply:
fixed

Reply:
I think the constants are OK, but FWIW, the k8s authors seem to be OK with using string literals in the instantiation of a GroupVersion. For example:
```sh
$ grep -r 'GroupVersion{Group: "", Version: "v1"}' | wc -l
914
```

Reply:
You might consider copying the cfg in this function too... It's kinda weird that a function named "createSecretAuth" would modify the argument you give it.

Reply:
Whenever you start talking about the principle of least surprise, I think of the law of surprise from the Witcher. No first born children should b exchanged here.

I can live with the hard coded strings, I'm just a little surprised they don't have something better.

Reply:
> Whenever you start talking about the principle of least surprise, I think of the law of surprise from the Witcher. No first born children should b exchanged here.
> 
> I can live with the hard coded strings, I'm just a little surprised they don't have something better.

LOL. I love the Witcher! :) (And no one wants my first-born, I promise you! :P) 

I was surprised too, I thought they'd have some exported var ready to use or something... and to be clear, I really don't mean this as a counter to your suggestion Terry, just I noticed that Steven was following the pattern of the authors.

Reply:
But now I have to ask, did you find any surprises when you returned home from Seattle?

Reply:
Ha! No surprises for me.